### PR TITLE
FUSETOOLS2-2389: UI test for Provide guidance when no folder is opene…

### PR DESCRIPTION
**Pull request by pospisilf**
…d and commands cannot work

### CODE REVIEW

File: src/ui-test/utils.ts [hunk 1]
```diff
@@ -602,3 +602,40 @@ export async function moveDebugBar(time: number = 60_000): Promise<void> {
...
+
+/**
+ * Checks if Notification Center contains notification with required text.
+ * 
+ * @param notificationText Text of notification. 
+ * @returns true if notification is present, false otherwise
+ */
+export async function notificationCenterContains(notificationText: string): Promise <boolean> {
+    const notifications = await new Workbench().getNotifications();
+    const length = await notifications.length;
```
Comment (djelinek) [inline src/ui-test/utils.ts:614]: seems like a good candidate for shorter `for` syntax

```typescript
const notifications = await new Workbench().getNotifications();
for(const notification of notifications) {
  const message = await notification.getMessage();
  if(message === notificationText){
	  return true;
  }
}
```

File: src/ui-test/tests/commands.cannot.work.test.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+ * limitations under the License.
+ */
+
+import path from "path";
+import { ActivityBar, EditorView, SideBarView, VSBrowser, WebDriver, Workbench } from "vscode-uitests-tooling";
+import { CAMEL_ROUTE_YAML_WITH_SPACE, CAMEL_RUN_ACTION_LABEL, CAMEL_RUN_DEBUG_ACTION_LABEL } from "../variables";
+import { expect } from "chai";
+import { notificationCenterContains, waitUntilNotificationShows } from "../utils";
+
+describe('Commands cannot work test', function () {
```
Comment (djelinek) [inline src/ui-test/tests/commands.cannot.work.test.ts:24]: I would go maybe with something more descriptive to the feature we test... `Check actions/commands requirements to run` or something like this.. WDYT?

File: src/ui-test/tests/commands.cannot.work.test.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        const explorerView = await activityBar.getViewControl('Explorer');
+        await explorerView.openView();
+
+        const explorer = await new SideBarView().getContent();
+        const sections = await explorer.getSections();
+
+        expect(await sections.at(0)?.getTitle()).to.equal("No Folder Opened");
+    });
+
+    it(`Click on run`, async function () {
```
Comment (djelinek) [inline src/ui-test/tests/commands.cannot.work.test.ts:53]: same here... `Click on 'run' button and check warning message is displayed`..

File: src/ui-test/tests/commands.cannot.work.test.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        await clickButtonAndVerifyNotification(CAMEL_RUN_ACTION_LABEL);
+    });
+    
+    it(`Click on debug`, async function () {
```
Comment (djelinek) [inline src/ui-test/tests/commands.cannot.work.test.ts:57]: same here... `Click on 'debug' button and check warning message is displayed`..

File: src/ui-test/tests/commands.cannot.work.test.ts [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+ */
+
+import path from "path";
+import { ActivityBar, EditorView, SideBarView, VSBrowser, WebDriver, Workbench } from "vscode-uitests-tooling";
+import { CAMEL_ROUTE_YAML_WITH_SPACE, CAMEL_RUN_ACTION_LABEL, CAMEL_RUN_DEBUG_ACTION_LABEL } from "../variables";
+import { expect } from "chai";
+import { notificationCenterContains, waitUntilNotificationShows } from "../utils";
+
+describe('Commands cannot work test', function () {
+    this.timeout(240000);
```
Comment (pospisilf) [inline src/ui-test/tests/commands.cannot.work.test.ts:25]: lower this value by a lot

File: src/ui-test/tests/commands.cannot.work.test.ts [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    
+    let driver: WebDriver;
+    let editorView: EditorView;
+
+    const NOTIFICATION_TEXT = "The action requires an opened folder/workspace to complete successfully.";
+
+    before(async function () {
+        await (new Workbench()).executeCommand("Workspaces: Close Workspace");
```
Comment (pospisilf) [inline src/ui-test/tests/commands.cannot.work.test.ts:33]: execute only if folder is opened

**Review by djelinek — APPROVED** [review]

**Pull request merged**